src/libraries/Microsoft.Extensions.Logging.Console/src/SystemdConsoleFormatter.cs

```diff
@@ -0,0 +1,132 @@
+            // systemd reads messages from standard out line-by-line in a '<pri>message' format.
+            // newline characters are treated as message delimiters, so we must replace them.
+            // Messages longer than the journal LineMax setting (default: 48KB) are cropped.
+            // Example:
+            // <6>ConsoleApp.Program[10] Request received
+
+            // loglevel
+            string logLevelString = GetSyslogSeverityString(logLevel);
+            textWriter.Write(logLevelString);
+
+            // timestamp
+            string timestampFormat = FormatterOptions.TimestampFormat;
+            if (timestampFormat != null)
+            {
+                DateTime dateTime = GetCurrentDateTime();
+                textWriter.Write(dateTime.ToString(timestampFormat));
+            }
+
+            // category and event id
+            textWriter.Write(category);
+            textWriter.Write('[');
+            textWriter.Write(eventId);
+            textWriter.Write(']');
+
+            // scope information
+            GetScopeInformation(textWriter, scopeProvider);
+
+            // message
+            if (!string.IsNullOrEmpty(message))
+            {
+                textWriter.Write(' ');
+                // message
+                WriteReplacingNewLine(textWriter, message);
+            }
+
+            // exception
+            // System.InvalidOperationException at Namespace.Class.Function() in File:line X
+            if (exception != null)
+            {
+                textWriter.Write(' ');
+                WriteReplacingNewLine(textWriter, exception.ToString());
+            }
+
+            // newline delimiter
+            textWriter.Write(Environment.NewLine);
+
+            static void WriteReplacingNewLine(TextWriter writer, string message)
+            {
+                string newMessage = message.Replace(Environment.NewLine, " ");
+                writer.Write(newMessage);
+            }
+        }
+
+        private DateTime GetCurrentDateTime()
+        {
+            return FormatterOptions.UseUtcTimestamp ? DateTime.UtcNow : DateTime.Now;
+        }
+
+        private static string GetSyslogSeverityString(LogLevel logLevel)
+        {
+            // 'Syslog Message Severities' from https://tools.ietf.org/html/rfc5424.
+            return logLevel switch
+            {
+                LogLevel.Trace => "<7>",
+                LogLevel.Debug => "<7>",        // debug-level messages
+                LogLevel.Information => "<6>",  // informational messages
+                LogLevel.Warning => "<4>",     // warning conditions
+                LogLevel.Error => "<3>",       // error conditions
+                LogLevel.Critical => "<2>",    // critical conditions
+                _ => throw new ArgumentOutOfRangeException(nameof(logLevel))
+            };
+        }
+
+        private void GetScopeInformation(TextWriter textWriter, IExternalScopeProvider scopeProvider)
```

Comment:
(nit) maybe rename this to `WriteScopeInformation`? "GetScopeInformation" sounds like it should return something.